internal/host/validator.go

```diff
@@ -625,8 +625,14 @@ func (v *validator) belongsToL2MajorityGroup(c *validationContext, majorityGroup
 		return ValidationPending
 	}
 
-	// TODO: Handle multple machine networks
-	return boolValue(funk.Contains(majorityGroups[string(c.cluster.MachineNetworks[0].Cidr)], *c.host.ID))
+	// TODO(mko) This rule should be revised as soon as OCP supports multiple machineNetwork
+	//           entries using the same IP stack.
```

Comment:
Is there an OCP issue tracking this work? How can we tell, 5 months from now, whether this was done or not?

Reply:
I'm not sure, it's still quite confusing to me which team exactly would be responsible for this. Maybe @williamcaban can provide some help in finding out whether it's already been requested ?

Reply:
I actually don't understand this TODO. Won't L2 validation always require all the machine networks to be available for all the hosts?

Reply:
>Won't L2 validation always require all the machine networks to be available for all the hosts?

Not all of them. In the future we want to be able to support the following configuration

```
machineNetwork
- 10.0.1.0/24
- 10.0.2.0/24
- 10.0.11.0/24
- 2001:db8::/32
```

In such a scenario one node will be 10.0.1.1 and the other 10.0.2.1 - so they do not belong to the same L2; but they create a valid setup so they should be able to belong to the same cluster. If we just blindly validate membership in every machineNetwork, we would forbid this scenario from happening